qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -145,73 +157,143 @@ def name(self):
 
     # ---------------------------------------------------------------------
     # Attributes
     # ---------------------------------------------------------------------
+    @deprecate_function(
+        "The Clifford.__getitem__ method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use getter of Clifford.paulis property instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def __getitem__(self, key):
         """Return a stabilizer Pauli row"""
-        return self._table.__getitem__(key)
+        return self.table.__getitem__(key)
 
+    @deprecate_function(
+        "The Clifford.__getitem__ method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.paulis property instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def __setitem__(self, key, value):
         """Set a stabilizer Pauli row"""
-        self._table.__setitem__(key, value)
+        self._paulis.__setitem__(key, value)
+
+    @property
+    def paulis(self):
+        """Return PauliList"""
+        return self._paulis
+
+    @paulis.setter
+    def paulis(self, value):
+        """Set the Pauli list"""
+        if not isinstance(value, PauliList):
+            self._paulis = PauliList(value)
+        else:
+            self._paulis = value.copy()
 
     @property
+    @deprecate_function(
+        "The Clifford.table method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.paulis method instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def table(self):
         """Return StabilizerTable"""
-        return self._table
+        return StabilizerTable(self.paulis.tableau(), phase=self.paulis.phase // 2)
 
     @table.setter
+    @deprecate_function(
+        "The Clifford.table method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.paulis method instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def table(self, value):
         """Set the stabilizer table"""
         # Note this setter cannot change the size of the Clifford
         # It can only replace the contents of the StabilizerTable with
         # another StabilizerTable of the same size.
         if not isinstance(value, StabilizerTable):
             value = StabilizerTable(value)
-        self._table._array[:, :] = value._table._array
-        self._table._phase[:] = value._table._phase
+        self.paulis = value
 
     @property
+    @deprecate_function(
+        "The Clifford.stabilizer method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.stabilizers method instead.",  # pylint:disable=bad-docstring-quotes
+    )
```

Comment:
Similar to above, `Clifford.stabilizer` is an attribute (as far as the user is concerned), though the new `Clifford.stabilizers` _is_ a method.  The same applies to the setter, and to `destabilizer` below.